src/frontend/mosh-server.cc

```diff
@@ -783,6 +796,28 @@ static void serve( int host_fd, Terminal::Complete &terminal, ServerConnection &
 	    connected_utmp = true;
 	  }
 	  #endif
+      #ifdef HAVE_SYSLOG
+	  if ( syslog_saved_addr_len != network.get_remote_addr_len()
+	       || memcmp( &syslog_saved_addr, &network.get_remote_addr(),
```

Comment:
My eyebrows went up at the use of `memcmp()` on what looked like a C++ object, but sadly I'm not sure we can actually do any better here, since `Addr` is actually a union of C structs.

The best thing would be to have `Addr` be a variant object with an `operator==`, but I'm not asking you to do that.

Reply:
This is the same as the utmp_connected case, minus the condition on utmp_connected.  If one changes both cases should.